tasks/rh-sign-image-cosign/rh-sign-image-cosign.yaml

```
@@ -77,15 +77,22 @@ spec:
         run_cosign () { # Expected arguments are [digest_reference, tag_reference]
             # Upload transparency log when rekor url is specified
             if [ -v REKOR_URL ]; then
-                COSIGN_COMMON_ARGS="-y --rekor-url=$REKOR_URL --key $SIGN_KEY"
+                COSIGN_COMMON_ARGS=(
+                  -y
+                  "--rekor-url=$REKOR_URL"
+                  "--key $SIGN_KEY"
+                )
             else
-                COSIGN_COMMON_ARGS="--tlog-upload=false --key $SIGN_KEY"
+                COSIGN_COMMON_ARGS=(
+                  "--tlog-upload=false"
+                  "--key $SIGN_KEY"
+                )
             fi
             echo "Signing manifest $1 ($2)"
             attempt=0
             until [ "$attempt" -gt "$(params.retries)" ] ; do # 3 retries by default
                 cosign -t 3m0s sign\
-                ${COSIGN_COMMON_ARGS}\
+                "${COSIGN_COMMON_ARGS[*]}"\
```

This would still have the same issue as the original change that broke it - it would still result in a single string. What you need is `@` instead of `*`.
```suggestion
                "${COSIGN_COMMON_ARGS[@]}"\
```

---

I tried that:

```
❌ Found ShellCheck warnings! /o\
error: Argument mixes string and array. Use * or separate argument.
   ┌─ tasks/rh-sign-image-cosign/rh-sign-image-cosign.yaml:97:18
   │
97 │                 "${COSIGN_COMMON_ARGS[@]}"\
   │                  ^^^^^^^^^^^^^^^^^^^^^^^^
   │
   = SC[21](https://github.com/konflux-ci/release-service-catalog/actions/runs/12414712984/job/34659400431?pr=752#step:4:22)45
   = Defect reference: https://github.com/koalaman/shellcheck/wiki/SC2145
```

---

And what do you think of that? :) I think shellcheck is wrong here. Look at the wiki. We're not mixing anything. Maybe try adding a space before \? That sounds like a good idea anyway.

---

If it is not a clean fix we can always disable shell check on this too. I haven't spent the time looking at this you others have, just throwing that out there. It wouldn't be our only instance of disabling it

---

adding a space made it happy. thanks